Dockerfile

```diff
@@ -1,10 +1,15 @@
-FROM node:16 AS build
-
+FROM registry.access.redhat.com/ubi8/nodejs-16:latest as builder
 ADD . /usr/src/app
 WORKDIR /usr/src/app
-RUN yarn install && yarn build
-
-FROM nginx:stable
+USER root
+RUN dnf config-manager --add-repo https://dl.yarnpkg.com/rpm/yarn.repo \
+ && dnf -y install yarn \
+ && yarn install \
+ && yarn build
 
-RUN chmod g+rwx /var/cache/nginx /var/run /var/log/nginx
-COPY --from=build /usr/src/app/dist /usr/share/nginx/html
+FROM registry.access.redhat.com/ubi8/nodejs-16:latest
+COPY --from=builder /usr/src/app/dist /opt/app-root/src/staticroot
+ENV VIEWS_DIR=/opt/app-root/src/staticroot
+ENV STATIC_DIR=/opt/app-root/src/staticroot
+ENV NODE_TLS_REJECT_UNAUTHORIZED="0"
+ENTRYPOINT ["node", "/opt/app-root/src"]
```

Comment:
Our plugin doesn't run a NodeJS server at all, I'm not sure we need the `VIEWS_DIR`, `STATIC_DIR`, `NODE_TLS_*` or even the `ENTRYPOINT`.

Does the UI load successfully without the nginx stuff here? The plugin pod hosts its webpack module on an nginx server for the console UI to load.

Reply:
I also don't think `/opt/app-root/src/staticroot` will work since the `STATIC_DIR` env variable is something that was handled explicitly in the NodeJS server code of mig-ui, which we don't have here.

Reply:
Understood. Let me see if I can just serve it with nginx

Reply:
I wonder if all you need is to revert the parts after the `yarn build` happens back to:
```
FROM nginx:stable
RUN chmod g+rwx /var/cache/nginx /var/run /var/log/nginx
COPY --from=build /usr/src/app/dist /usr/share/nginx/html
```

Reply:
This works with the ubi8 nginx-120 image. Docs are at:
https://github.com/sclorg/nginx-container/blob/master/1.20/README.md#direct-usage-with-a-mounted-directory

The 'big' thing using the UBI image gets us is it is an easier frame of reference to adapt for releng when they build downstream. They won't be able to use `nginx:stable` or `node:16` to build/ship.

```
FROM registry.access.redhat.com/ubi8/nginx-120:latest
COPY --from=build /usr/src/app/dist /opt/app-root/src
ENTRYPOINT ["nginx", "-g", "daemon off;"]
```

Reply:
Another important one is that pulling docker images in quay tends to cause failures due to dockers rate limiting.

Reply:
Ah, that makes sense!